src/os/bluestore/BlueStore.cc

```diff
@@ -2003,6 +2003,29 @@ void BlueStore::ExtentMap::update(KeyValueDB::Transaction t,
   }
 }
 
+decltype(BlueStore::Blob::id) BlueStore::ExtentMap::allocate_spanning_blob_id()
+{
+  if(spanning_blob_map.empty())
+    return 0;
+  decltype(BlueStore::Blob::id) bid = spanning_blob_map.rbegin()->first + 1;
+  // bid is valid and available.
+  if(bid >= 0 && spanning_blob_map.find(bid) == spanning_blob_map.end())
```

Comment:
the key in a map is ordered, so the largest key + 1, if not wrapped, is  deem to be not represented in the map?